File: leetcodePractice/6:11:2020/winSum.py
def winSum(nums, k):
	if not nums or len(nums) < k:
		return []

	result = []
	j, window_sum = 0, 0
	for i in range(len(nums)):
		while j < len(nums) and j - i < k:
			window_sum += nums[j]
			j += 1
		if j - i == k:
			result.append(window_sum)
		window_sum -= num[i]
	
	return result

File: leetcodePractice/7-12-2020/winSum.java
/**
 * 使用同向双指针模版解决滑动窗口问题
 */

public int[] winSum(int[] nums, int k) {
    if (nums == null || nums.length < k) {
        return new int[]{};
    }
    if (k == 0) {
        return new int[]{nums.length};
    }

    int[] results = new int[nums.length];
    int j = 0, sum = 0;
    for (int i = 0; i < nums.length; i++) {
        while (j < nums.length && j - i < k) {
            sum += nums[j];
            j++;
        } 
        if (j - i == k) {
            results[i] = sum;
        }
        sum -= nums[i];
    }

    return results;
}